examples/tree-view/src/Node/treeSlice.js

```diff
@@ -0,0 +1,67 @@
+import { createSlice } from "@reduxjs/toolkit";
+import generateTree from "../generateTree";
+
+const { tree, idCount } = generateTree();
+
+let nextId = idCount;
+const initialState = tree;
+
+const getAllDescendants = (tree, id) => {
+  const descendants = [id];
+  const run = (id) => {
+    tree[id].childIds.forEach((childId) => {
+      descendants.push(childId);
+      run(childId);
+    });
+  };
+
+  run(id);
+  return descendants;
+};
+
+const deleteMany = (tree, ids) => {
+  ids.forEach((id) => delete tree[id]);
+};
+
+const treeSlice = createSlice({
+  name: "tree",
+  initialState,
+  reducers: {
+    increment(state, action) {
+      const nodeId = action.payload;
+      state[nodeId].counter++;
+    },
+    createNode(state) {
+      nextId++;
```

Comment:
:exclamation: This is a side effect, and shouldn't happen in a reducer. But, if you add a `prepare` callback for this reducer, we can do the exact same line in the prepare callback and then put the new ID value into the action itself.